File: backend/app/core/openai_client.py
import os
from openai import OpenAI


def get_openai_client() -> OpenAI:
    """
    Returns a configured OpenAI client.

    This should be the ONLY place where the OpenAI client is constructed.
    All jobs and services should import and use this function.
    """
    api_key = os.getenv("OPENAI_API_KEY")
    if not api_key:
        raise RuntimeError("OPENAI_API_KEY is not set")

    return OpenAI(api_key=api_key)

File: frontend/app/lib/openaiClient.ts
/**
 * Shared OpenAI client factory.
 * Validates the API key and returns a configured client.
 * Used by askService and discoverService.
 */

import OpenAI from 'openai';
import { serviceUnavailable } from '@/app/lib/errors';

/**
 * Returns a configured OpenAI client.
 * Throws 503 ServiceUnavailable if the API key is not set.
 */
export function getOpenAIClient(): OpenAI {
  const apiKey = process.env.OPENAI_API_KEY;
  if (!apiKey) {
    throw serviceUnavailable('OpenAI API key not configured');
  }
  return new OpenAI({ apiKey });
}
